suite-common/wallet-core/src/discovery/discoveryThunks.ts

```diff
@@ -74,44 +73,32 @@ export const filterUnavailableAccountTypes = (
             !device?.unavailableCapabilities?.[networkAccount.accountType!], // exclude by account types (ex: taproot)
     );
 
-/**
- * @deprecated Old implementation, temporarily duplicated here for suite, while the new one is used in suite-native
- * Fully replaced by combined use of filterUnavailableNetworks and filterUnavailableAccountTypes
- */
-export const filterUnavailableNetworksCompatible = (
-    enabledNetworks: NetworkSymbol[],
-    device?: TrezorDevice,
-) =>
-    networksCompatibility.filter(n => {
-        const firmwareVersion = getFirmwareVersion(device);
-        const internalModel = device?.features?.internal_model;
-
-        const isSupportedInSuite =
-            !n.support || // support is not defined => is supported
-            !internalModel || // typescript. device undefined. => supported
-            (n.support[internalModel] && // support is defined for current device
-                versionUtils.isNewerOrEqual(firmwareVersion, n.support[internalModel])); // device version is newer or equal to support field in networks => supported
-
-        return (
-            isTrezorConnectBackendType(n.backendType) && // exclude accounts with unsupported backend type
-            enabledNetworks.includes(n.symbol) &&
-            !n.isHidden &&
-            !device?.unavailableCapabilities?.[n.accountType!] && // exclude by account types (ex: taproot)
-            !device?.unavailableCapabilities?.[n.symbol] && // exclude by network symbol (ex: xrp on T1B1)
-            isSupportedInSuite
-        );
-    });
-
 const calculateProgress =
     (discovery: Discovery) =>
     (_dispatch: any, getState: any): PartialDiscovery => {
-        const { numberOfNonCardano, numberOfCardano } = discovery.networks.reduce(
-            (acc, symbol) => {
-                if (symbol === 'ada') {
-                    acc.numberOfCardano += 1;
-                } else {
-                    acc.numberOfNonCardano += 1;
-                }
+        const device = selectDevice(getState());
```

Comment:
what's going on here? See [L741, L748](https://github.com/trezor/trezor-suite/blob/ced3f2a8a54c2636010c61e1f11be1bce03c692c/suite-common/wallet-core/src/discovery/discoveryThunks.ts#L741-L748) where it's first called.

**Previously**, `networksSymbols` from `filterUnavailableNetworksCompatible` was non-unique, [like in comment above](https://github.com/trezor/trezor-suite/pull/14398/files/d8e95808ac13344515846ca1d08255ad06b957a4..ced3f2a8a54c2636010c61e1f11be1bce03c692c#r1764541695), except here **it actually mattered**, and imho the previous implementation was kinda occult.
In `calculateProgress`, we need the _number_ of loaded accounts, so the number of duplicated `discovery.networks` symbols was counted.

**Now**, `discovery.networks` is unique, and it is responsibility of `calculateProgress` to iterate through accounts.

:bulb:  Note how `filterUnavailableNetworksCompatible` is split between `filterUnavailableNetworks` and `filterUnavailableAccountTypes`.